test/test_pytree.py

```diff
@@ -294,25 +305,39 @@ def test_broadcast_to_and_flatten(self):
                     LeafSpec(),
                 ]),
             ]),
-        ]), "L(T(*,*,L(*,*)))"),
-    ], name_fn=lambda _, str_spec: str_spec)
-    def test_pytree_serialize(self, spec, str_spec):
-        self.assertEqual(pytree_to_str(spec), str_spec)
-        self.assertTrue(spec == str_to_pytree(str_spec))
-        self.assertTrue(spec == str_to_pytree(pytree_to_str(spec)))
+        ]),
+    ],)
+    def test_pytree_serialize(self, spec):
+        self.assertTrue(spec == treespec_loads(treespec_dumps(spec)))
 
     def test_pytree_serialize_namedtuple(self):
         Point = namedtuple("Point", ["x", "y"])
         spec = TreeSpec(namedtuple, Point, [LeafSpec(), LeafSpec()])
-        str_spec = "N(Point(x, y),*,*)"
 
-        self.assertEqual(pytree_to_str(spec), str_spec)
-
-        roundtrip_spec = str_to_pytree(pytree_to_str(spec))
+        roundtrip_spec = treespec_loads(treespec_dumps(spec))
```

Comment:
Should check that the result of treespec_dumps is actually a str lol.
Also, to detect BC issues, we should treespec_dumps something complicated into a string that we check into the codebase and then add a test that treespec_loads is able to load that string